### PR TITLE
Don't schedule next step if it's null [MAILPOET-4450]

### PR DESCRIPTION
**Pull request by alex-mailpoet**
[MAILPOET-4450]

[MAILPOET-4450]: https://mailpoet.atlassian.net/browse/MAILPOET-4450?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Automation/Engine/Control/StepRunner.php [hunk 1]
```diff
@@ -116,6 +116,11 @@ private function runStep(array $args): void {
       return;
     }
 
+    // no need to schedule a new step if the next step is null
+    if (empty($nextStepArgs[0]['step_id'])) {
```
Comment (websupporter) [inline mailpoet/lib/Automation/Engine/Control/StepRunner.php:120]: I think you need to set the status of the workflowrun to `WorkflowRun::STATUS_COMPLETE` which currently happens in L91 when no stepId was found.

Comment (alex-mailpoet) [reply mailpoet/lib/Automation/Engine/Control/StepRunner.php:120]: Good catch! I've added completion of the workflow run here.

**Review by websupporter — APPROVED** [review]

**Comment by websupporter** [comment]
Thank you @alex-mailpoet. LGTM

**Comment by JanJakes** [comment]
I think this one is also good to merge.

**Pull request merged**